lib/encode.js

```diff
@@ -13,39 +21,61 @@ import {inspect} from './util.js';
  *   CBOR-LD bytes.
  * @param {documentLoaderFunction} options.documentLoader -The document loader
  *   to use when resolving JSON-LD Context URLs.
- * @param {boolean} [options.compressionMode=1] - `1` to use compression mode
- *   version 1, `0` to use no compression.
- * @param {Map} [options.appContextMap] - A map of JSON-LD Context URLs and
- *   their encoded CBOR-LD values (must be values greater than 32767 (0x7FFF)).
- * @param {diagnosticFunction} [options.diagnose] - A function that, if
- * provided, is called with diagnostic information.
- *
+ * @param {number} options.varintValue - The varint value for the
+ *  registry entry associated with the resulting CBOR-LD payload. For legacy
+ *  support, use varintTagValue = -1.
+ * @param {Map} options.keywordsTable -A map of JSON-LD keywords
+ *  and their associated CBOR-LD integer values.
+ * @param {Map} options.stringTable - A map of string values and
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.urlSchemeTable - A map of URL schemes to
+ *  their associated CBOR-LD integer values.
+ * @param {Map} options.typedLiteralTable - A map of JSON-LD types
+ *  to maps that map values of that type to their associated CBOR-LD
+ *  integer values.
+ * @param {diagnosticFunction} options.diagnose - A function that, if
+ *  provided, is called with diagnostic information.
+ * @param {Map} options.appContextMap - For use with the legacy value of
+ *  varintTagValue.
+ * @param {number} options.compressionMode - For use with the legacy value of
+ *  varintValue.
  * @returns {Promise<Uint8Array>} - The encoded CBOR-LD bytes.
  */
 export async function encode({
-  jsonldDocument, documentLoader, appContextMap = new Map(),
-  compressionMode = 1, diagnose
+  jsonldDocument, documentLoader, varintValue = 'legacy',
+  keywordsTable = new Map(),
+  stringTable = new Map(),
+  urlSchemeTable = new Map(),
+  typedLiteralTable = new Map(),
+  diagnose,
+  appContextMap = new Map(),
+  compressionMode = 1
 } = {}) {
-  if(!(compressionMode === 0 || compressionMode === 1)) {
-    throw new TypeError(
-      '"compressionMode" must be "0" (no compression) or "1" ' +
-      'for compression mode version 1.');
```

Comment:
I think we still need this for backwards compatibility -- and it should additionally check whether `legacy` is set for `varintValue`. So we need this back, but we can move it below and throw if it was passed and `varintValue !== 'legacy'`. That means we'll also want to not set a default value for it in the function so we can detect whether it was actually passed or not (`undefined`). The default value can be set to `1` when we see that `varintValue === 'legacy'`.